src/main/java/scratch/UCERF3/utils/MFD_InversionConstraint.java

```diff
@@ -132,6 +133,11 @@ public void write(JsonWriter out, MFD_InversionConstraint value) throws IOExcept
 				Region region = new Region(value.region);
 				regionAdapter.write(out, region);
 			}
+
+			if(value instanceof MFD_WeightedInversionConstraint){
```

Comment:
@voj @kevinmilner  I think it's not great to have a parent class be 'aware' of it's subclass(es). Ideally the subclass would override the parent class and mutate the json returned from calling the superclass method, but that's not  possible with this API (where the method returns void).

Reply:
@chrisbc I agree that the broken class hierarchy is not great. We could move the Adapter class into its own file and then that would be fixed.